### PR TITLE
msm state save

### PR DESCRIPTION
**Pull request by RomanLevkovych**
Saving the last state  that was sent to EM in persistent storage, so we could resume our work correctly if system unexpectedly died.

### CODE REVIEW

File: applications/ExecutionManager/include/execution_manager.hpp [hunk 1]
```diff
@@ -78,6 +82,9 @@ class ExecutionManager
   std::mutex m_activeProcessesMutex;
   std::set<ProcName> m_activeProcesses;
 
+	std::mutex m_failedAppsMutex;
+	std::set<ProcName> m_failedApps;
```
Comment (ihor96berizka) [inline applications/ExecutionManager/include/execution_manager.hpp]: wrong indentations

File: applications/ExecutionManager/include/msm_handler.hpp [hunk 2]
```diff
@@ -13,10 +13,12 @@ class MsmHandler
   bool registerMsm(pid_t processId, const std::string& appName);
   bool checkMsm(pid_t processId) const;
   pid_t msmPid() const;
+	bool exists() const; 
+	void clearMsm();
```
Comment (ihor96berizka) [inline applications/ExecutionManager/include/msm_handler.hpp]: indentation

File: applications/ExecutionManager/src/execution_manager.cpp [hunk 3]
```diff
@@ -26,6 +26,7 @@ ExecutionManager::ExecutionManager(
         std::unique_ptr<ExecutionManagerClient::IExecutionManagerClient> client)
     : m_appHandler{std::move(applicationHandler)},
       m_activeProcesses{},
+			m_failedApps{},
```
Comment (ihor96berizka) [inline applications/ExecutionManager/src/execution_manager.cpp]: same as prev 2 comments

File: applications/ExecutionManager/src/execution_manager.cpp [hunk 4]
```diff
@@ -322,4 +327,16 @@ void ExecutionManager::confirmComponentState
...
+	std::lock_guard<std::mutex> lk{m_failedAppsMutex};
+	m_failedApps.insert(app);
+}
+
+void ExecutionManager::removeFailedApp(const ProcName& app)
+{
+	std::lock_guard<std::mutex> ls{m_activeProcessesMutex};
+	m_activeProcesses.erase(app);
+}
+
```
Comment (ihor96berizka) [inline applications/ExecutionManager/src/execution_manager.cpp]: where are these methods used?

Comment (ihor96berizka) [reply applications/ExecutionManager/src/execution_manager.cpp]: Also, looks like these methods are related to another task(about handling of failed apps)

File: applications/MachineStateManager/test/msm_test.cpp [hunk 5]
```diff
@@ -24,6 +25,8 @@ class MsmTest : public ::testing::Test
            std::make_unique<MachineStateClientMock>();
    std::unique_ptr<ExecutionManager::ManifestReaderMock> manifestReaderMock =
            std::make_unique<NiceMock<ExecutionManager::ManifestReaderMock>>();
+   std::unique_ptr<NiceMock<KeyValueStorageMock>> storageMock =
+           std::make_unique<NiceMock<KeyValueStorageMock>>();
```
Comment (ulianaStashkiv) [inline applications/MachineStateManager/test/msm_test.cpp]: Use there StrictMock.

File: applications/MachineStateManager/src/machine_state_manager.cpp [hunk 6]
```diff
@@ -84,7 +93,7 @@ StateError MachineStateManager::registerMsm(const std::string& applicationName)
 
 std::string MachineStateManager::getNewState()
 {
-  std::string newState = m_newStatesProvider->getData();
+  std::string newState = getNewStateForStartRun();
 
   while(m_availableStates.cend() ==
```
Comment (ihor96berizka) [inline applications/MachineStateManager/src/machine_state_manager.cpp]: Why do you use while loop here?
m_availableStates is updated only in ctor,
So, if getNewStateForStartRun() returns invalid state - you will stuck in this loop.

**Pull request merged**